assets/admin/editor-wizard/steps/patterns-step.js

```diff
@@ -25,17 +25,28 @@ const { Fill, Slot } = createSlotFill( 'Patterns Upsell' );
  * @param {Function} props.onCompletion On completion callback.
  */
 const PatternsStep = ( { title, replaces, onCompletion } ) => {
-	const { resetEditorBlocks } = useDispatch( editorStore );
+	const { resetEditorBlocks, editPost } = useDispatch( editorStore );
 	const logEvent = useLogEvent();
+	const editorSettings = useSelect( ( select ) =>
+		select( editorStore ).getEditorSettings()
+	);
+	const availableTemplates = editorSettings.availableTemplates;
 
-	const onChoose = ( blocks, name ) => {
+	const onChoose = ( blocks, name, template ) => {
 		const newBlocks = replaces
 			? replacePlaceholders( blocks, replaces )
 			: blocks;
 
 		resetEditorBlocks( newBlocks );
 		onCompletion();
 
+		const templateSlug =
+			template && Object.keys( availableTemplates ).includes( template )
+				? template
+				: '';
```

Comment:
I think we should not call the edited post if the template is not available, to avoid any kind of issue and extra savings.

Reply:
Thanks @gabrielcaires , yeah it is for when the user picks a pattern from the pattern selection window while creating course. I've updated the code with a comment and changed the code to only edit post in case a template is selected [here](https://github.com/Automattic/sensei/pull/6200/commits/20412f66e98e6cbadffe19c547952e80fbeefa05). Can you kindly take a look again? Thanks